### PR TITLE
Feat 47 craniotomy2019

### LINKED ISSUE #47: Map 2019 sharepoint database for Craniotomy
# User story
As a user, I want to map the 2019 sharepoint database to the aind-data-schema Craniotomy model, so I can have a more complete Procedures instance.

# Acceptance criteria
- [x] add examples of craniotomies to excel
- [x] Map all known procedure fields
- [x] Map known fields for (standard) Craniotomy (craniotomy_coordinates_ml, craniotomy_coordinates_ap, craniotomy_size, dura_removed, workstation_id)
- [x] handle validation errors

# Sprint Ready Checklist 
- [x] 1. Acceptance criteria defined
- [x] 2. Team understands acceptance criteria
- [x] 3. Team has defined solution / steps to satisfy acceptance criteria
- [x] 4. Acceptance criteria is verifiable / testable
- [x] 5. External / 3rd Party dependencies identified
- [x] 6. Ticket is prioritized and sized

# Notes
missing fields: implant_part_number, protective_material, anaesthesia duration

### PR DESCRIPTION
**Pull request by mekhlakapoor**
closes #47 
missing fields: implant_part_number, protective_material, anaesthesia duration

### CODE REVIEW

**Comment by mekhlakapoor** [comment]
{"message":"Valid Model.","data":{"describedBy":"https://raw.githubusercontent.com/AllenNeuralDynamics/aind-data-schema/main/src/aind_data_schema/procedures.py","schema_version":"0.5.2","subject_id":"436083","headframes":[{"type":null,"start_date":"2019-01-07T08:00:00Z","end_date":"2019-01-11T08:00:00Z","experimenter_full_name":"Jun Zhuang","iacuc_protocol":null,"animal_weight":null,"anaesthesia":null,"notes":null,"headframe_material":null,"well_part_number":null,"well_type":null}],"craniotomies":[{"type":null,"start_date":"2019-01-09","end_date":"2019-01-09","experimenter_full_name":"Jun Zhuang","iacuc_protocol":"1804","animal_weight":"21.3","anaesthesia":{"type":"isoflurane","level":"1.5"},"notes":null,"craniotomy_hemisphere":"Left","craniotomy_coordinates_ml":2.8,"craniotomy_coordinates_ap":1.3,"craniotomy_size":5.0,"implant_part_number":null,"dura_removed":true,"protective_material":null,"workstation_id":"SWS 6"}],"mri_scans":null,"injections":[{"type":null,"start_date":"2019-01-09","end_date":"2019-01-09","experimenter_full_name":"Jun Zhuang","iacuc_protocol":"1804","animal_weight":"21.3","anaesthesia":{"type":"isoflurane","duration":null,"level":"Select..."},"notes":null,"injection_materials":null,"recovery_time":null,"injection_duration":10,"workstation_id":"SWS 6","instrument_id":"NJ#6","injection_hemisphere":"Left","injection_coordinate_ml":-2.3,"injection_coordinate_ap":-2.3,"injection_coordinate_depth":2.6,"injection_angle":0.0,"injection_type":"Nanoject (Pressure)","injection_volume":"200 nL"}],"fiber_implants":null,"water_restrictions":null,"training_protocols":null,"tissue_preparations":null,"other_procedures":null,"notes":null}}%

File: src/aind_metadata_service/sharepoint/client.py [hunk 1]
```diff
@@ -44,6 +47,11 @@ class ProcedureType(Enum):
         INJECTION = "Injection"
         INJ = "INJ"
         OPTIC_FIBER_IMPLANT = "Optic Fiber Implant"
+        HP_TRANSCRANIAL = "HP Transcranial (for ISI)"
```
Comment (dyf) [inline src/aind_metadata_service/sharepoint/client.py:50]: @saskiad Is there anything valuable to keep track of out of these values? Looks mostly like project codes so we can probably ignore. The fact that it's a transcranial craniotomy is maybe already covered by craniotomy size?

**Review by dyf — APPROVED** [review]
Looks good.  @saskiad I see they they have some meta

File: src/aind_metadata_service/sharepoint/client.py (hunk 1 above)
Comment (mekhlakapoor) [reply src/aind_metadata_service/sharepoint/client.py:50]: I'll actually be mapping HP Transcranial to headplant in another PR

Comment (saskiad) [reply src/aind_metadata_service/sharepoint/client.py:50]: yes, I think these values will help us to map to protocols. There are also hidden parameters associated with these procedures that don't show up in the request form (eg. craniotomy coordinates) that are determined by the type of procedure. 
Some of the items on this list might be obsolete for our purposes, but we can prune them as we need

File: src/aind_metadata_service/sharepoint/client.py [hunk 2]
```diff
@@ -44,6 +47,11 @@ class ProcedureType(Enum):
...
+        WHC_NP = "WHC NP"
```
Comment (saskiad) [inline src/aind_metadata_service/sharepoint/client.py:51]: Can this map to "Whole hemisphere craniotomy Neuropixels" - lots of acronyms aren't always helpful

**Review by saskiad — CHANGES_REQUESTED** [review]
see comments

File: src/aind_metadata_service/sharepoint/client.py [hunk 3]
```diff
@@ -44,6 +47,11 @@ class ProcedureType(Enum):
         INJECTION = "Injection"
         INJ = "INJ"
```
[the remark this answers is not in the document]
Comment (mekhlakapoor) [reply src/aind_metadata_service/sharepoint/client.py:48]: In the sharepoint, an injection procedure type can be demarked by different labels (sometimes its "Injection", "INJ", or "Stereotaxic Injection") and its usually "INJ" when its an entry with multiple procedures (ex: HP+INJ+C CAM). These 3 all end up mapping to injection but I'm using the procedure type enum to define the procedure types found in the sharepoint

File: src/aind_metadata_service/sharepoint/client.py [hunk 4]
```diff
@@ -44,6 +47,11 @@ class ProcedureType(Enum):
...
         OPTIC_FIBER_IMPLANT = "Optic Fiber Implant"
+        HP_TRANSCRANIAL = "HP Transcranial (for ISI)"
+        WHOLE_HEMISPHERE_CRANIOTOMY_NP = "WHC NP"
```
Comment (saskiad) [inline src/aind_metadata_service/sharepoint/client.py:51]: shouldn't the spelled out part be the other side - the part in quotes? The acronym from sharepoint on the left and what it means on the right? Or do I not understand how you are using this?

Comment (mekhlakapoor) [reply src/aind_metadata_service/sharepoint/client.py:51]: the right part in quotes is how its labeled in sharepoint

**Review by saskiad — APPROVED** [review]

**Pull request merged**